src/loader.ts

```diff
@@ -93,4 +103,61 @@ export default class Loader {
   clearAbortTimeout() {
     clearTimeout(this.abortTimeoutId);
   }
+
```

Comment:
Felt a bit weird putting this post code in loader, but they share some helper functions. Happy to refactor.

Reply:
Yeah, it does feel a bit awkward. I'm 👍 for a refactor here if it isn't a pain.

Reply:
Sounds good, will do